### ISSUE TITLE
Font file remains open after reading?

### ISSUE THREAD

**Issue by alxx** [ruby2d/ruby2d#191]
I have a suspicion that once ruby2d reads the font, it neglects to close the file; this means that I'm currently unable to create more than 239 `Text` instances:

```ruby
(['Abc'] * 239).each do |t|
  Text.new(t, font: __dir__ + '/../fonts/arial.ttf')
end
```

The code above works fine, but if I replace 239 with 240, I get the error `Error Domain=NSPOSIXErrorDomain Code=24 "Too many open files"`

I suggest that in `ext/ruby2d/text.c` at line 29 after opening the font with `TTF_OpenFont` it should also be closed with `TTF_CloseFont`, as per instructions here: https://www.libsdl.org/projects/SDL_ttf/docs/SDL_ttf_18.html#SEC18

I'd try it locally myself, but I really don't know how to overwrite a C file in a gem.

**Comment by mariovisic** [comment]
Thanks @alxx 

Looking over the code in `text.c` it does indeed open a new font file per text, so with a large amount of text objects you will run into this issue. We can't easily `free` it as the font is used on each draw call.

It's possible to open the font only when we need to draw, then close it again but I don't think this would be very efficient.

Instead a better solution may be to use the same font instance for multiple text objects. We have a card on the roadmap to implement a media manager which should solve this in a nice way 👍 

https://github.com/ruby2d/ruby2d/projects/1#card-56850298

**Comment by alxx** [comment]
Thanks for looking into this. Looking forward for the update.

**Comment by blacktm** [comment]
Fixed in #204
